### PR TITLE
Clippy lint fixes

### PR DESCRIPTION
**Pull request by jb55**
Continues from #24 so review that first.

Ones that confused me the most:

```
lint: remove noop drops
lint: silence dubious clippy warning about unsafe blocks
```

perhaps those indicate real issues but I'm not that familiar with unsafe block symantics

### CODE REVIEW

File: src/wallet.rs [hunk 1]
```diff
@@ -219,12 +219,12 @@ impl Wallet {
...
-            rpc: rpc,
+            network,
+            rpc,
             mnemonic: mnemonic.to_owned(),
-            master_xpriv: master_xpriv,
-            external_xpriv: external_xpriv,
-            internal_xpriv: internal_xpriv,
+            master_xpriv,
+            external_xpriv,
+            internal_xpriv,
```
Comment (stevenroose) [inline src/wallet.rs]: Hmm, I don't really like this "optimization".. It's clearly convoluting a variable name and a field name which are two different concepts that just happen to share the same name.

Could we `#[allow(clippy::redundant_field_names)]` perhaps?

File: src/network.rs [hunk 2]
```diff
@@ -161,7 +161,7 @@ impl Network {
         &NETWORKS
     }
 
-    pub fn get(id: &String) -> Option<&'static Network> {
+    pub fn get(id: &str) -> Option<&'static Network> {
```
Comment (stevenroose) [inline src/network.rs:164]: Hmm, I remember doing this and feeling annoyed by it. Something complained about `&str` when I did this. If that seems to be solved, great! :)

File: src/session.rs [hunk 3]
```diff
@@ -103,7 +103,6 @@ impl SessionManager {
         if !self.sessions.remove(&sess) {
             throw!("session is unmanaged");
         }
-        unsafe { drop(&*sess) };
```
Comment (stevenroose) [inline src/session.rs]: Hmm, are you sure no drop of some kind is required here?

File: src/wally.rs [hunk 4]
```diff
@@ -230,9 +230,10 @@ pub fn tx_get_elements_signature_hash(
     sighash: u32,
     segwit: bool,
 ) -> sha256d::Hash {
-    let flags = match segwit {
```
Comment (stevenroose) [inline src/wally.rs:233]: Heh, yeah I do that because it's a line shorter and the logic is trivial. Anyhow, this is ok.

**Review by stevenroose — CHANGES_REQUESTED** [review]

File: src/session.rs (hunk 3 above)
Comment (jb55) [reply src/session.rs]: yeah this is the one I was dubious about, the linter claims this is a no-op. I don't know enough about rust FFI / memory management to comment on this yet. I'll drop the commit for now and keep the lint warning/error

**Comment by jb55** [comment]
v2: git range-diff 0b9b27b...654708b updates:

- removed e14b099 `lint: cleanup redunant field names in struct initializations`
- removed a56e192 `lint: remove noop drops`
- added 654708b `lint: allow redunant field names in wallet.rs`

**Comment by stevenroose** [comment]
reACK 654708b

**Pull request merged**